.github/workflows/ci.yaml

```diff
@@ -0,0 +1,316 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      - master
+      - release-*
+  push:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  ### Configuration checks ###
+  helm-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install Helm
+        uses: azure/setup-helm@v4.2.0
+        id: install
+      - run: helm lint deployments/gpu-operator/
+  validate-csv:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get Golang version
+        id: vars
+        run: |
+          GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
```

Comment:
Is there no way to reuse this and eliminate the duplication?

Reply:
I think the use top-level `env` field of the workflow.yaml to set global env vars.

In this case, I would prefer hardcoding the golang version in the top-level `env field`. While automated fetching of golang version is a plus, I feel this would be a good tradeoff for the sake of simplicity

Reply:
I am not too familiar with GH actions, so I am not sure if we can fetch the golang version once and reuse that throughout this file. 

It's a tradeoff, but I think I prefer duplicating this logic so that `versions.mk` remains the single source of truth for the Go version.

Reply:
Ideally we would have some shared worklflow definitions that we can use to pull in logic such as this.